firebase-perf/src/main/java/com/google/firebase/perf/application/FragmentStateMonitor.java

```diff
@@ -60,12 +60,44 @@ public void onFragmentResumed(@NonNull FragmentManager fm, @NonNull Fragment f)
     super.onFragmentResumed(fm, f);
     // Start Fragment screen trace
     logger.debug("FragmentMonitor %s.onFragmentResumed", f.getClass().getSimpleName());
+    Trace fragmentTrace =
+        new Trace(getFragmentScreenTraceName(f), transportManager, clock, appStateMonitor);
+    fragmentTrace.start();
+
+    if (f.getParentFragment() != null) {
+      fragmentTrace.putAttribute(
+          Constants.PARENT_FRAGMENT_ATTRIBUTE_KEY,
+          f.getParentFragment().getClass().getSimpleName());
+    }
+    if (f.getActivity() != null) {
+      fragmentTrace.putAttribute(
+          Constants.ACTIVITY_ATTRIBUTE_KEY, f.getActivity().getClass().getSimpleName());
+    }
+
+    fragmentToTraceMap.put(f, fragmentTrace);
   }
 
   @Override
   public void onFragmentPaused(@NonNull FragmentManager fm, @NonNull Fragment f) {
     super.onFragmentPaused(fm, f);
     // Stop Fragment screen trace
     logger.debug("FragmentMonitor %s.onFragmentPaused ", f.getClass().getSimpleName());
+    if (!fragmentToTraceMap.containsKey(f)) {
+      logger.error(
```

Comment:
Is this an error or a warning? In this case, I don't think the developer can act on that error. So, I guess this qualifies for a warning.

Reply:
Changed to `logger.warn`